Hexastore.Rocks/RocksGraphProvider.cs

```diff
@@ -39,61 +52,93 @@ public RocksGraphProvider(ILogger<RocksGraphProvider> logger, string path = null
             .IncreaseParallelism(6)
             .SetCompression(Compression.No);
             */
-
-            _db = RocksDb.Open(options, dataPath);
-            _logger.LogInformation($"created rocksdb at {dataPath}");
         }
 
+
         public bool ContainsGraph(string id, GraphType type)
         {
             var key = MakeKey(id, type);
-            return _db.Get(key) == "1";
+            var db = GetOrOpenDB(id);
```

Comment:
It could be useful to have another GetDb function that returns null if there is no Db. In that case ContainsGraph would also return false.

Reply:
Good catch, there are some scenarios here that should be a check only. I'll fix this.